### PR TITLE
CarouselHeader Component

### PR DESCRIPTION
**Pull request by chadfawcett**
Screen Shot 2020-10-14 at 11 59 29 AM

* I spent a bit of time working on how best to import all the icons as they currently are, but ultimately figured it'd be better to just pass the icon source as a prop. Ideally we'll get svgs for the icons which will make the importing better, so didn't want to spend too much time implementing a way to import all the pngs.
* Added `@testing-library/react-native`

Resolves [pe1107](https://app.clubhouse.io/parks/story/1107/carousel-header-component)

### CODE REVIEW

**Comment by chrisfosterelli** [comment]
@chadfawcett I'm not technically assigned dev on this project, but given Kaileen is away today, would a review from me be helpful to get this approved? Or do you need Kaileen's eyes on something?

**Comment by chadfawcett** [comment]
@chrisfosterelli A review wouldn't hurt, but it's not blocking.

File: src/components/CarouselHeader.test.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
+import React from 'react'
+import { render } from '@testing-library/react-native'
+import CarouselHeader from './CarouselHeader'
+
+test('Should render using props', () => {
+  const { getByText } = render(
+    <CarouselHeader
+      title="Test Title"
+      subheading="Test subheading"
+      icon={require('../images/icons/activity/hiking.png')}
```
Comment (chrisfosterelli) [inline src/components/CarouselHeader.test.js:10]: **question (non-blocking)**: Can this be an import at the top of the file?

**Review by chrisfosterelli — APPROVED** [review]
Good to go!

File: src/components/CarouselHeader.test.js (hunk 1 above)
Comment (chadfawcett) [reply src/components/CarouselHeader.test.js:10]: Probably, this is just the way the docs show it being done.

I didn't focus too much time on it as it's likely going to change.

Comment (chadfawcett) [reply src/components/CarouselHeader.test.js:10]: Oh whoops, this is the test file. Yeah, doing it once at the top of the file probably makes more sense for when/if there are more than one test.

**Review by mccullochk — APPROVED** [review]
This looks good @chadfawcett. I just have one comment. To me it feels like the park icons are a property of `<CarouselHeader />` component itself and should be controlled by it, not the parent component. That would diminish the usability of this component in other contexts but to me the component is specifically a park icon a title and subtitle which may or may not be fixed (will they all be specifically 25km away?). Even if the distance isn't fixed, the variable that changes is the distance not the whole subtitle. So the input I see coming into `<CarouselHeader />` being `{ title, type, distance }`. It could probably be simplified even more to just having the `type` as the variable and having the words `Great ${type} near you` fixed as well. 🤷‍♀️

It's a small difference so maybe not worth a lengthy discussion.

**Comment by chadfawcett** [comment]
@mccullochk I originally thought that you'd pass in a type for the icon as well, but with the icons in their current png format it added a lot of extra work which I felt wasn't worth investing in since the SVGs are on their way. It also simplifies the logic of the component itself. being more of a view only component.

I wanted to leave the title and subtitle as regular string props as I didn't know how set in stone they were. I think it's better to leave them more customizable by the parent component.

**Comment by chadfawcett** [comment]
I'll merge in for now as it's been approved, but I'm open to changing the props more once we sort out the svg icons.

**Pull request merged**